Wangscape/utils.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <type_traits>
+#include <stdexcept>
+
+namespace
+{
+
+template<typename I>
+I _pow(I base, I exp)
+{
+    I result = 1;
+    for (int i = 0; i < std::numeric_limits<I>::digits; i++)
+    {
+        if (exp & 1)
+            result *= base;
+        else if (exp == 0)
```

Comment:
The `for` condition is meant to prevent the shifted sign bit from being used in `exp & 1`. But the wrapping function already ensures `exp > 0` here, so `while` can be used.